src/main/java/it/smartcommunitylab/aac/identity/service/IdentityProviderService.java

```diff
@@ -148,6 +148,10 @@ private void setProviderProperties(ProvidersProperties providers) {
                     cip.setSettings(settings.getConfiguration());
                     cip.setConfiguration(configuration);
 
+                    // add temporary version as it might be required by jdbc repository persistence
+                    int registeredVersion = (int) (System.currentTimeMillis() / (60*1000)); // current time in minutes
+                    cip.setVersion(registeredVersion);
```

Comment:
you need to set  the version only if the field is null: the user could have put the version in the config and thus you should respect that.

Reply:
Added commit to add version only if not defined from config.  otherwise keep what was written in application.yaml.
The previous behaviour was to always ignore the version suggested by the user. It was not clear whether this change would also let the user define the version, given it is not something that previously possible.